snapshotter/snapshotter.py

```diff
@@ -378,6 +394,13 @@ def snapshot(source, dest, debug=False, min_snapshots=3, extra_args=None):
     """
     date = _datetime()
     user, host, snapshots_root = _parse_path(dest)
+    if max_snapshots > -1 and max_snapshots - min_snapshots < 0:
+        raise InconsistentArgumentsError(
+            "--max-snapshots must be greater than --min-snapshots")
+    if max_snapshots > -1 and max_snapshots + 1 - min_snapshots >= 0:
+        snapshots = _ls_snapshots(dest)
+        for i in range(len(snapshots) - max_snapshots + 1):
+            _rm(snapshots[i], user, host, directory=True, debug=debug)
```

Comment:
Nice idea. I didn't saw the function `_remove_oldest_snapshot`. It's not the less memory efficient but it's shorter and easily understandable. If we keep the test with min-snapshots and max-snapshots juste before then it should work without problems.